tests/assembly/simd-intrinsic-mask-load.rs

```diff
@@ -0,0 +1,63 @@
+// verify that simd masked load does not introduce additional bit shift operations
+//@ revisions: x86-avx x86-avx512
+//@ [x86-avx] compile-flags: --target=x86_64-unknown-linux-gnu -C llvm-args=-x86-asm-syntax=intel
+//@ [x86-avx] compile-flags: -C target-feature=+avx
+//@ [x86-avx] needs-llvm-components: x86
+//@ [x86-avx512] compile-flags: --target=x86_64-unknown-linux-gnu -C llvm-args=-x86-asm-syntax=intel
+//@ [x86-avx512] compile-flags: -C target-feature=+avx512f,+avx512vl,+avx512bw,+avx512dq
+//@ [x86-avx512] needs-llvm-components: x86
+//@ assembly-output: emit-asm
+//@ compile-flags: --crate-type=lib -O
+
+#![feature(no_core, lang_items, repr_simd, intrinsics)]
+#![no_core]
+#![allow(non_camel_case_types)]
+
+// Because we don't have core yet.
+#[lang = "sized"]
+pub trait Sized {}
+
+#[lang = "copy"]
+trait Copy {}
+
+#[repr(simd)]
+pub struct f32x8([f32; 8]);
+
+#[repr(simd)]
+pub struct m32x8([i32; 8]);
+
+#[repr(simd)]
+pub struct f64x4([f64; 4]);
+
+#[repr(simd)]
+pub struct m64x4([i64; 4]);
+
+extern "rust-intrinsic" {
+    fn simd_masked_load<M, P, T>(mask: M, pointer: P, values: T) -> T;
+}
+
+// CHECK-LABEL: load_f32x8
+#[no_mangle]
+pub unsafe fn load_f32x8(mask: m32x8, pointer: *const f32, output: *mut f32x8) {
+    // x86-avx-NOT: vpslld
+    // x86-avx: vmovaps ymm0
+    // x86-avx-NEXT: vmaskmovps
+    // x86-avx512-NOT: vpslld
+    // x86-avx512: vpcmpgtd k1
```

Comment:
I initially expected `vpmovd2m` here, llvm seems to prefer the compare instruction since it can have a memory operand. So this should be unlikely to change in an llvm update.